core/src/main/java/org/apache/calcite/util/format/FormatElementEnum.java

```diff
@@ -165,10 +173,17 @@ public enum FormatElementEnum implements FormatElement {
       return String.format(Locale.ROOT, "%02d", calendar.get(Calendar.SECOND));
     }
   },
+  MS("The millisecond as a decimal number (000-999)") {
+    @Override public String format(Date date) {
+      final Calendar calendar = Work.get().calendar;
+      calendar.setTime(date);
+      return String.format(Locale.ROOT, "%03d", calendar.get(Calendar.MILLISECOND));
+    }
+  },
   TZR("The time zone name") {
     @Override public String format(Date date) {
       // TODO: how to support timezones?
-      throw new UnsupportedOperationException();
+      return "";
```

Comment:
why we return empty string for this case

Reply:
We do so in order to match PostgreSQL's output when formatting `timestamp`s: https://dbfiddle.uk/VkeR6jUV